src/server/pkg/hashtree/db.go

```diff
@@ -49,6 +49,9 @@ var (
 	SentinelByte = []byte{'*'}
 )
 
+// Filter is a function for filtering hashtree keys.
+type Filter func(k []byte) (bool, error)
+
```

Comment:
Are there any implementations of this type that actually return an error. I did a quick check and didn't see it, if not could we remove the `error` return argument?

Reply:
I think the original idea I had with this was that we could have a range filter and the error would be used for breaking out of the iteration, but I don't know if that will be too useful with the indexing we have now. I will go ahead and remove the error return for now, and we can re-visit later if necessary.